src/pedal.cpp

```diff
@@ -58,13 +58,19 @@ int Pedal::GenerateMIDI(FunctorParams * functorParams)
     if (!HasDir()) return FUNCTOR_CONTINUE;
 
     int velocity;
+    int tickOffset = 0;
     switch (GetDir())
     {
     case pedalLog_DIR_down:
         velocity = 126;
         break;
     case pedalLog_DIR_up:
         velocity = 0;
+
+        // if you have a pedal event that ends right as another begins (ex. one in the end of measure 1 and start of measure 2), 
+        // they will overlap because they occur at the same midi time
+        // so we just add a -1 offset to the ending time to remedy this until a better method is thought of
```

Comment:
I think that I have fixed the problem of ordering pedal up/down messages in the midifile library.  The `MidiFile::sortTracks()` function should now sort pedal on/offs occurring at the same time so that the pedal off is placed before the pedal on: 

https://github.com/craigsapp/midifile/blob/bb1e75842625ca75010407753758e48c12b02a78/src-library/MidiEventList.cpp#L584-L599

An alternate method would be to use the `MidiEvent::seq` variable to force a specific ordering of the pedal messages.  A lower `MidiEvent::seq` value would force an event to be sorted before a higher valued `seq`:

https://github.com/craigsapp/midifile/blob/9d7e042be14eab9a70ef81c2b5267af88053b268/src-library/MidiEventList.cpp#L548-L555

But your solution should also work well.